frontend/src/Frontend/UI/Dialogs/Signing.hs

```diff
@@ -154,30 +154,45 @@ uiQuickSign
   -> (QuickSignRequest, Either Text QuickSignResponse -> JSM ())
   -> Event t ()
   -> m (mConf, Event t ())
-uiQuickSign ideL (qsr, writeSigningResponse) _onCloseExternal = (mempty, ) <$> do
-  if _quickSignRequest_commands qsr == []
-     then sendResp =<< failWith "QuickSign request was empty" ""
-     else case partitionEithers $ fmap csdToSigData $ _quickSignRequest_commands qsr of
-       ([], payloads) ->
-         quickSignModal ideL sendResp payloads
-       (es, _) -> sendResp =<<
-         failWith ("QuickSign request contained invalid commands:\n" <> T.unlines (map T.pack es)) ""
+uiQuickSign ideL (qsr, writeSigningResponse) onCloseExternal = (mempty, ) <$> do
+  let sendResp = performEvent . fmap (liftJSM . writeSigningResponse . Right) -- We use Either for legacy reasons, but the quicksign protocol now allows for error handling in its response, so we just `const Right` now
+      emptyReqResponse = do
+        eClose <- failureModal "QuickSign request was empty" "" onCloseExternal
+        sendResp $ QSR_Error QuicksignError_EmptyList <$ eClose
+  if _quickSignRequest_csds qsr == []
+    then emptyReqResponse
+    else case partitionEithers $ fmap snd sdReqs of
+       ([], reqs) -> quickSignModal ideL sendResp reqs onCloseExternal
+       otherwise -> do
```

Comment:
Does this type check?  Seems like it shouldn't given https://hackage.haskell.org/package/base-4.17.0.0/docs/Prelude.html#v:otherwise

Reply:
Yes, though i can change it to `_`